Fix spelling of function name

mogemimi opened the pull request:
This commit fixes three small spelling mistakes:
- `relaseTextures()` -> `releaseTextures()`
- `caculateFinalPositionWithRelativeWidget()` -> `calculateFinalPositionWithRelativeWidget()`
- `caculateFinalPositionWithRelativeAlign()` -> `calculateFinalPositionWithRelativeAlign()`

Thanks for your work on this engine. :relaxed:

mogemimi committed:
Fix spelling of function name

pandamicro commented:
We also appreciate your contribution ~ :+1:

ricardoquesada commented:
this might break backwards compatibility, but +1 for adding this patch

pandamicro commented:
Only protected function and private functions have been modified, it will cause problem if someone extend FontAtlas in C++ and call `relaseTexture` function.

I think it's pretty safe to merge this PR, and make it clear in our API documentation

mogemimi commented:
@pandamicro @ricardoquesada Thank you for reviewing!
If you like, could you please merge this? :blush:

ricardoquesada commented:
hey, what about adding a new method that is called `relaseTextures` (the old name) that calls the new one `releaseTextures()` and we tag as deprecate?

That way it will be 100% backwards compatible.

mogemimi committed:
Mark the old functions as CC_DEPRECATED_ATTRIBUTE

mogemimi commented:
Hi, it sounds like a good idea to me! :+1:
I just added the old functions marked as deprecated, please see 12ece31.
Also I've rebased on the current master, thanks.

mogemimi commented:
[ci rebuild]

ricardoquesada commented:
thanks. merged
